rollup/fees/rollup_fee.go

```diff
@@ -60,8 +60,16 @@ func EstimateL1DataFeeForMessage(msg Message, baseFee, chainID *big.Int, signer
 		return nil, err
 	}
 
-	l1BaseFee, overhead, scalar := readGPOStorageSlots(rcfg.L1GasPriceOracleAddress, state)
-	l1DataFee := calculateEncodedL1DataFee(raw, overhead, l1BaseFee, scalar)
+	l1BaseFee, overhead, scalar, l1BlobBaseFee, commitScalar, blobScalar := readGPOStorageSlots(rcfg.L1GasPriceOracleAddress, state)
+
+	var l1DataFee *big.Int
+
+	if !config.IsCurie(blockNumber) {
+		l1DataFee = calculateEncodedL1DataFee(raw, overhead, l1BaseFee, scalar)
+	} else {
+		l1DataFee = calculateEncodedL1DataFeeCurie(raw, l1BaseFee, l1BlobBaseFee, commitScalar, blobScalar)
+	}
```

Comment:
A future note: it would've been nice if this calculation was in the contract itself. Might be inefficient tho.

Reply:
The problem is that the contract does not have access to the raw transaction data. But it is in the contract: https://github.com/scroll-tech/scroll/pull/1343/files#diff-946456f6b3ce3deda3ba31e88f4d629b5f94a93e0369c9f85465135d74961827R264 Technically we could call into the EVM here, but it would complicate things a bit.